worldmap.js

```diff
@@ -117,11 +116,95 @@ module.exports = function(RED) {
             }
             node.status({});
         });
-        sockets[this.path].on('connection', callback);
+        sockets[node.path].on('connection', callback);
+    }
+
+    var WorldMap = function(n) {
+        worldMap(this, n);
     }
     RED.nodes.registerType("worldmap",WorldMap);
 
 
+    function HTML(ui, config) {
+        var width = config.width;
+        if (width == 0) {
+            var group = RED.nodes.getNode(config.group);
+            if (group) {
+                width = group.config.width;
+            }
+        }
+        var height = config.height;
+        if (height == 0) {
+            height = 10;
+        }
+        var size = ui.getSizes();
+        var frameWidth = (size.sx +size.cx) *width -size.cx -10;
+        var frameHeight = (size.sy +size.cy) *height -size.cy -10;
+        var url = encodeURI(config.path);
+        var html = `
+<div>
+    <iframe src="${url}" width="${frameWidth}px" height="${frameHeight}px" style="border: none;"></iframe>
+</div>
+`;
+        return html;
+    }
+
+    function checkConfig(node, conf) {
+        if (!conf || !conf.hasOwnProperty("group")) {
+            node.error("no group");
+            return false;
+        }
+        return true;
+    }
+
+    var ui = undefined;
+    
+    function UIWorldMap(config) {
+        try {
+            var node = this;
+            if(ui === undefined) {
+                ui = RED.require("node-red-dashboard")(RED);
+            }
+            worldMap(node, config);
+            var done = null;
+            if (checkConfig(node, config)) {
+                var html = HTML(ui, config);
+                done = ui.addWidget({
+                    node: node,
+                    order: config.order,
+                    group: config.group,
+                    width: config.width,
+                    height: config.height,
+                    format: html,
+                    templateScope: "local",
+                    emitOnlyNewValues: false,
+                    forwardInputMessages: false,
+                    storeFrontEndInputAsState: false,
+                    convertBack: function (value) {
+                        return value;
+                    },
+                    beforeEmit: function(msg, value) {
+                        return { msg: { items: value } };
+                    },
+                    beforeSend: function (msg, orig) {
+                        if (orig) { return orig.msg; }
+                    },
+                    initController: function($scope, events) {
+                    }
+                });
+            }
+         }
+        catch (e) {
+            console.log(e);
+        }
+        node.on("close", function() {
+            if (done) {
+                done();
+            }
+        });
+    }
+    RED.nodes.registerType("ui_worldmap", UIWorldMap);
```

Comment:
just wondering if we could make this conditional... i.e. don't register if dashboard isn't installed...

Reply:
hmm - well not there - but...  let me think